xcell/cls/data.py

```diff
@@ -184,52 +275,34 @@ def get_tracers_bare_name_pair(self, tr1, tr2, connector='-'):
         trreq = connector.join([tr1_nn, tr2_nn])
         return trreq
 
-    def _get_cl_trs_names(self, wsp=False):
-        cl_tracers = []
-        tr_names = self.get_tracers_used(wsp)
-        for i, tr1 in enumerate(tr_names):
-            for tr2 in tr_names[i:]:
-                trreq = self.get_tracers_bare_name_pair(tr1, tr2)
-                trreq_inv = self.get_tracers_bare_name_pair(tr2, tr1)
-                if (trreq not in self.data['cls']) and \
-                   (trreq_inv in self.data['cls']):
-                    # If the inverse is in the data file, reverse it
-                    # internally
-                    self.data['cls'][trreq] = self.data['cls'][trreq_inv]
-                    del self.data['cls'][trreq_inv]
-                elif (trreq not in self.data['cls']) and \
-                     (trreq_inv not in self.data['cls']):
-                    # If still not present, skip
-                    continue
-                clreq = self.data['cls'][trreq]['compute']
-                if clreq == 'all':
-                    pass
-                elif (clreq == 'auto') and (tr1 != tr2):
-                    continue
-                elif clreq == 'None':
-                    continue
-                cl_tracers.append((tr1, tr2))
-        return cl_tracers
-
     def get_cl_trs_names(self, wsp=False):
         lab = ['no_wsp', 'wsp'][wsp]
         if self.cl_tracers[lab] is None:
-            self.cl_tracers[lab] = self._get_cl_trs_names(wsp)
-        return self.cl_tracers[lab]
+            cl_tracers = []
+            tr_names = self.get_tracers_used(wsp)
+            tmat = self.get_tracer_matrix()
 
-    def _get_cov_trs_names(self, wsp=False):
-        cl_tracers = self.get_cl_trs_names(wsp)
-        cov_tracers = []
-        for i, trs1 in enumerate(cl_tracers):
-            for trs2 in cl_tracers[i:]:
-                cov_tracers.append((*trs1, *trs2))
+            for tr1 in tr_names:
+                for tr2 in tr_names:
+                    req = tmat[(tr1, tr2)]
+                    if req['inv'] or (req['compute'] is False):
+                        continue
+                    cl_tracers.append((tr1, tr2))
 
-        return cov_tracers
+            self.cl_tracers[lab] = cl_tracers
+
+        return self.cl_tracers[lab]
 
     def get_cov_trs_names(self, wsp=False):
         lab = ['no_wsp', 'wsp'][wsp]
```

Comment:
add docstring as for the previous method

Reply:
Done